### PR TITLE
implement sending WATCHDOG=1 notification messages from systemd itself

### PR DESCRIPTION
**Pull request by dpiegdon**

### CODE REVIEW

**Comment by evverx** [comment]
Well, I don't know how do you test it:

```
../src/shared/watchdog.c: In function ���watchdog_ping���:
../src/shared/watchdog.c:131:17: error: ���reutrn��� undeclared (first use in this function)
                 reutrn sd_notify(0, "WATCHDOG=1");
                 ^
```

Can you provide something like [that](https://github.com/systemd/systemd/pull/2907)?
(Just to be sure that your patch works as expected and doesn't break something)

**Comment by dpiegdon** [comment]
we are developing an embedded linux based on bitbake/oe-core jethro, which uses systemd v225.
(thats also why the typo slipped in, sorry about that)
so right now i can't provide further info for _this_ patch, sorry. i will try to setup a test-box with systemd-master in the coming days if need be.

**Comment by evverx** [comment]
> i will try to setup a test-box with systemd-master if need be

Ok.
The simplest way:

```
# install qemu, strace
$ cd test/TEST-01-BASIC
$ sudo make clean setup
$ sudo make run KERNEL_APPEND=systemd.unit=multi-user.target
```

Works on `Ubuntu 16.04`, `Fedora 23`

File: src/core/main.c [hunk 1]
```diff
@@ -1666,8 +1666,27 @@ int main(int argc, char *argv[]) {
...
+                                arg_runtime_watchdog = new_time;
+                                log_debug(PACKAGE_STRING " setting systemd user slice watchdog time to %d us", arg_runtime_watchdog);
+                                watchdog_set_timeout(&arg_runtime_watchdog);
+                        } else {
+                                arg_runtime_watchdog = 0;
+                        }
+                } else {
+                        arg_runtime_watchdog = 0;
+                }
+        }
```
Comment (poettering) [inline src/core/main.c:1690]: Please have a look at CODING_STYLE, there are many issues with the coding style used here, please make sure to submit code that is in line with CODING_STYLE.

That said, I am pretty sure we should simply use sd-event's watchdog support for this. i.e. sd-event already can send out WATCHDOG=1 notification messages on its own, see sd_event_set_watchdog(3) man page, and we really should use this here, too.

**Comment by dpiegdon** [comment]
in about 2 weeks i will take care of this. vacation :)

**Pull request closed without merging**